### ISSUE TITLE
Rogue 'f' in Machine Name for some agents

### ISSUE THREAD

**Issue by leesoh** [EmpireProject/Empire#344]
## Empire Version

2.0_beta
## OS Information (Linux flavor, Python version)

elementaryOS 0.4, Debian 8
## Expected behavior and description of the error, including any actions taken immediately prior to the error. The more detail the better.

When running list on agents, my sample size of one Windows 10 box has an 'f' prepended to it's hostname
## Screenshot of error, embedded text output, or Pastebin link to the error

```
Name            Lang  Internal IP     Machine Name    Username            Process             Delay    Last Seen
---------       ----  -----------     ------------    ---------           -------             -----    --------------------
MZ6D9SP1        ps    192.168.<snip> fW10BOX          W10BOX\user01       powershell/5572     5/0.0    2016-10-02 08:18:15
```
## Additional Information

The host in question is actually my desktop, which is running the VM with Empire installed. The Empire box and W10BOX are both on Subnet1, which is just the NAT interface, but the Internal IP listed is W10BOX's LAN IP. I suspect this is a red herring, but thought it worth mentioning.

**Comment by xorrior** [comment]
Definitely seems odd. I'll take a look.

**Comment by leesoh** [comment]
I can't imagine how this would matter, but the machine in question is the Type 2 hypervisor for the system running Empire. (Windows 10 > VMware Workstation > [Debian | Kali | elementaryOS] > Empire 2.0_beta)

**Comment by leesoh** [comment]
I suspect this is an overflow of the **Internal IP** value.

When I run `sysinfo` on a host with an f, I see the following:
```
Listener:         https://192.168.1.143:8443
Internal IP:      192.168.35.2 fe80::bc0b:f9d8:1c03:54b5 8:: 192.168.1.1 fe80::c569:e9da:27a:ddb1 
Username:         DOMAIN\username
Hostname:         HOSTNAME01
OS:               Microsoft Windows 7 Enterprise 
High Integrity:   0
Process Name:     powershell
Process ID:       14664
Language:         powershell
Language Version: 5
```
When I run `sysinfo` from another guest, I get this:
```
Listener:         https://192.168.1.143:8443
Internal IP:      192.168.1.141
Username:         DMZ\Administrator
Hostname:         WS01
OS:               Microsoft Windows 7 Professional 
High Integrity:   1
Process Name:     powershell
Process ID:       3096
Language:         powershell
Language Version: 2
```
For some reason the **Internal IP** value for a host to guest includes all addresses, while guest to guest includes just a single address.

**Comment by enigma0x3** [comment]
@leesoh, if you are able to reproduce this, can you do an ipconfig /all on the host that you are seeing this from? I would like to try to verify that the character at the beginning of the hostname is the last character/number of an IP address assigned to the host.

**Comment by leesoh** [comment]
@enigma0x3 It's the first character of the IPv6 address, not the last of any other address associated with the system. I believe @rvrsh3ll was working on this and found a solution (column one, Internal IP, was grabbing the first 18 characters of the IP address, not 16.

https://github.com/adaptivethreat/Empire/blob/654bc8c915eaa4ea5f53e743c961e9fcf743e69b/lib/common/agents.py#L942

**Comment by xorrior** [comment]
This should be resolved in the Empire 2.0_beta and master branch.
